testutils/src/main/java/com/amplifyframework/testutils/sync/SynchronousAuth.java

```diff
@@ -252,11 +254,52 @@ public Object confirmResetPassword(
      */
     @NonNull
     public AuthSession fetchAuthSession() throws AuthException {
-        return Await.<AuthSession, AuthException>result(AUTH_OPERATION_TIMEOUT_MS, (onResult, onError) ->
-                asyncDelegate.fetchAuthSession(onResult, onError)
+        return Await.result(AUTH_OPERATION_TIMEOUT_MS, asyncDelegate::fetchAuthSession);
+    }
+
+    /**
+     * Remembers current device synchronously.
+     * @return Dummy object - just indicates it completed successfully
+     * @throws AuthException exception
+     */
+    public Object rememberDevice() throws AuthException {
+        return Await.<Object, AuthException>result(AUTH_OPERATION_TIMEOUT_MS, (onResult, onError) ->
+                asyncDelegate.rememberDevice(() -> onResult.accept(new Object()), onError)
+        );
+    }
+
+    /**
+     * Forgets the current device synchronously.
+     * @return Dummy object - just indicates it completed successfully
+     * @throws AuthException exception
+     */
+    public Object forgetDevice() throws AuthException {
```

Comment:
Naming: this should be `forgetCurrentDevice()`, to clearly distinguish it from `forgetDevice(AuthDevice specificDevice)`. I would not compromise the naming just to match iOS. We should ask the iOS team to update their name if they're not using this, too.

Reply:
They have it set up such that `AuthDevice` parameter is nullable, and if none is passed, the method exhibits above behavior. It was in my discretion to overload the method and make `AuthDevice` a `@NonNull` parameter. I believe it is okay to rename this method as you suggest.